sections/synths/components/RedeemableDeprecatedSynthsModal.tsx

```diff
@@ -0,0 +1,361 @@
+import { FC, useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { useTranslation, Trans } from 'react-i18next';
+import { Synths } from '@synthetixio/contracts-interface';
+import Wei, { wei } from '@synthetixio/wei';
+import useSynthetixQueries, { Balances, DeprecatedSynthBalance } from '@synthetixio/queries';
+import { Svg } from 'react-optimized-image';
+
+import BaseModal from 'components/BaseModal';
+import { ButtonTransaction } from 'components/Form/common';
+
+import { FlexDivColCentered, FlexDivCentered, BlueStyledExternalLink } from 'styles/common';
+import media from 'styles/media';
+
+import Connector from 'containers/Connector';
+import GasSelector from 'components/GasSelector';
+import TxConfirmationModal from 'sections/shared/modals/TxConfirmationModal';
+import { ModalContent, ModalItemTitle, NoTextTransform } from 'styles/common';
+import PendingConfirmation from 'assets/svg/app/pending-confirmation.svg';
+import { formatCryptoCurrency, formatCurrency } from 'utils/formatters/number';
+import TxState from 'sections/gov/components/TxState';
+import Currency from 'components/Currency';
+import { GreyHeader } from 'sections/gov/components/common';
+
+const RedeemDeprecatedSynthsModal: FC<{
+	redeemAmount: Wei;
+	redeemableDeprecatedSynthsAddresses: string[];
+	redeemBalances: DeprecatedSynthBalance[];
+	synthBalances: Balances | null;
+	onDismiss: () => void;
+	onRedeemConfirmation: (txHash: string | null) => void;
+}> = ({
+	onDismiss,
+	onRedeemConfirmation,
+	redeemAmount,
+	redeemableDeprecatedSynthsAddresses,
+	redeemBalances,
+	synthBalances,
+}) => {
+	const { t } = useTranslation();
+	const { synthetixjs } = Connector.useContainer();
+
+	const [gasPrice, setGasPrice] = useState<Wei>(wei(0));
+	const [txModalOpen, setTxModalOpen] = useState<boolean>(false);
+
+	const sUSDBalance = synthBalances?.balancesMap[Synths.sUSD]?.balance ?? wei(0);
+	const Redeemer = synthetixjs!.contracts.SynthRedeemer;
+
+	const { useContractTxn } = useSynthetixQueries();
+	const txn = useContractTxn(Redeemer, 'redeemAll', [redeemableDeprecatedSynthsAddresses], {
+		gasPrice: gasPrice.toBN(),
+	});
+
+	useEffect(() => {
+		switch (txn.txnStatus) {
+			// case 'prompting':
```

Comment:
probably better to handle the modal appearing here instead of handleRedeem

Reply:
fixed